addons/rating/models/rating_mixin.py

```diff
@@ -232,13 +238,39 @@ def rating_apply(self, rate, token=None, rating=None, feedback=None, subtype_xml
             else:
                 subtype_id = self.env['ir.model.data']._xmlid_to_res_id(subtype_xmlid)
             feedback = tools.plaintext2html(feedback or '')
-            self.message_post(
-                author_id=rating.partner_id.id or None,  # None will set the default author in mail_thread.py
-                body="<img src='%s' alt=':%s/5' style='width:18px;height:18px;float:left;margin-right: 5px;'/>%s"
-                % (rating.rating_image_url, rate, feedback),
-                rating_id=rating.id,
-                subtype_id=subtype_id,
+
+            scheduled_date = (
+                fields.Datetime.now() + datetime.timedelta(hours=2)
+                if delay_sending else None
+            )
+
+            rating_body = (
+                markupsafe.Markup("<img src='%s' alt=':%s/5' style='width:18px;height:18px;float:left;margin-right: 5px;'/>%s")
+                % (rating.rating_image_url, rate, feedback)
             )
+
+            if rating.message_id:
+                # update the existing message
+                if any(mail.state != 'outgoing' for mail in rating.message_id.mail_ids):
+                    raise UserError(_('The rating is already sent, you can not change it'))
+
+                rating.message_id.body = rating_body
```

Comment:
What about introducing a ``def _message_update_message`` method that would allow to update a message and update the related mails at mail.thread level ? That way it can be tested (notably with ACLs in mind) and rating would just use the feature.

Reply:
There is a ``_message_update_content_after_hook`` method called when updating message directly from discuss to call in our new method if we do it (and maybe update directly ``_update_content`` on mail.message to call the new method on the thread itself, as I don't like having that kind of method on mail.message).